keps/sig-windows/20190103-windows-node-support.md

```diff
@@ -0,0 +1,196 @@
+- CRIs other than Dockershim: CRI-containerd support is forthcoming
+
+### What will never work (without underlying OS changes)
+- Certain Pod functionality
+    - Privileged containers
+    - Reservations are not enforced by the OS, but overprovisioning could be blocked with `--enforce-node-allocatable=pods` (pending: tests needed)
+    - Certain volume mappings
+      - Single file & subpath volume mounting
+      - Host mount projection
+      - DefaultMode (due to UID/GID dependency)
+      - readOnly root filesystem. Mapped volumes still support readOnly
+    - Termination Message - these require single file mappings
+- CSI plugins, which require privileged containers
+- [Some parts of the V1 API](https://github.com/kubernetes/kubernetes/issues/70604)
+- Overlay networking support in Windows Server 1803 is not fully functional using the `win-overlay` CNI plugin. Specifically service IPs do not work on Windows nodes. This is currently specific to `win-overlay` - other CNI plugins (OVS, AzureCNI) work.
+
+### Relevant resources/conversations
+
+- [sig-architecture thread](https://groups.google.com/forum/#!topic/kubernetes-sig-architecture/G2zKJ7QK22E)
+- [cncf-k8s-conformance thread](https://lists.cncf.io/g/cncf-k8s-conformance/topic/windows_conformance_tests/27913232)
+- [kubernetes/enhancements proposal](https://github.com/kubernetes/features/issues/116)
+
+
+### Risks and Mitigations
+
+**Second class support**: Kubernetes contributors are likely to be thinking of Linux-based solutions to problems, as Linux remains the primary OS supported. Keeping Windows support working will be an ongoing burden potentially limiting the pace of development. 
+
+**User experience**: Users today will need to use some combination of taints and node selectors in order to keep Linux and Windows workloads separated. In the best case this imposes a burden only on Windows users, but this is still less than ideal.
+
+## Graduation Criteria
+
+
+## Implementation History
+
+
+## Testing Plan
+
+
+### Test Dashboard
+
+All test cases will be built in kubernetes/test/e2e, scheduled through [prow](github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml), and published on the [TestGrid SIG-Windows dashboard](https://testgrid.k8s.io/sig-windows) daily. This will be the master list of what needs to pass to be declared stable and will include all tests tagged [SIG-Windows] along with the subset of conformance tests that can pass on Windows.
+
+
+Additional dashboard pages will be added over time as we run the same test cases with additional CRI, CNI and cloud providers. They reflect work that may be stabilized in v1.15 or later and is not strictly required for v1.14.
+
+- Windows Server 2019 on GCP - this is [in progress](https://k8s-testgrid.appspot.com/google-windows#windows-prototype), but not required for v1.14
+- Windows Server 2019 with OVN+OVS & Dockershim
```

Comment:
Are these additional dashboards? Are they all required other than GCP? What configuration will be shown on the main dashboard?

Reply:
Will clarify in the next update on #732